src/librustc/ty/mod.rs

```diff
@@ -2977,7 +2988,7 @@ fn associated_item_def_ids<'a, 'tcx>(tcx: TyCtxt<'a, 'tcx, 'tcx>,
                           .map(|id| tcx.hir.local_def_id(id.node_id))
                           .collect()
         }
-        hir::ItemKind::TraitAlias(..) => vec![],
+        hir::ItemKind::TraitAlias(..) => todo,
```

Comment:
This should just return `vec!`. Trait aliases do not define any associated items themselves. The associated item resolution code *does* have a notion of walking the superpredicates when doing lookups, I would think we should build on that.

Reply:
Makes sense.